lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -124,10 +124,16 @@ export default function BasicLtiLaunchApp({ clientRpc }) {
       setErrorState('error-fetching-canvas-file');
     } else if (
       e instanceof ApiError &&
-      e.errorCode === 'canvas_file_not_found_in_course'
+      e.errorCode !== null &&
+      [
+        'canvas_file_not_found_in_course',
+        'canvas_group_set_not_found',
+        'canvas_group_set_empty',
+        'canvas_student_not_in_group',
+      ].includes(e.errorCode)
     ) {
       setError(e);
-      setErrorState('canvas-file-not-found-in-course');
+      setErrorState(/** @type {ErrorState} */ (e.errorCode.replace(/_/g, '-')));
```

Comment:
We might not merge anything like this but I was not going to let the TS compiler beat me

Reply:
Heh ;) - This is a completely reasonable solution for "trust me, I know this is OK" situations like this.